### PR TITLE
Feature/cmp initializer and integration test sample

### PR DESCRIPTION
**Pull request by alextremp**
This PR:

* unifies the CMP initialization for both real and test usage
* add the test repositories for integration tests
* add a save new consent test
* fixes integration problems in used use cases
* add an custom event polyfill in order to dispatch the cmpReady event

### CODE REVIEW

File: src/cmp/infrastructure/createEvent.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* eslint-disable no-undef */
+import './event-polyfill'
```
Comment (victuxbb) [inline src/cmp/infrastructure/createEvent.js]: Maybe this should be added as cross sites polyfill...

https://github.com/SUI-Components/sui/blob/master/packages/sui-polyfills/src/index.js

File: test/cmp/infrastructure/factory/IABConsentFactoryTest.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import IABConsentFactory from '../../../../src/cmp/infrastructure/factory/IABCon
 describe('IABConsentFactory', () => {
   describe('createConsent', () => {
     it('Should return a ConsentString with a globalVendorList', done => {
-      const givenGlobalVendorList = require('./../../resources/globalvendorlist.json')
+      const givenGlobalVendorList = require('../../../resources/globalvendorlist.json')
```
[the remark this answers is not in the document]
Comment (alextremp) [reply test/cmp/infrastructure/factory/IABConsentFactoryTest.js]: 👍 Doing!

File: src/cmp/infrastructure/cmpInitializer.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  storeConsentGlobally = false,
+  gdprApplies = true,
+  vendorConsentsFactory,
+  consentFactory,
+  vendorListRepository,
+  consentRepository,
+  vendorConsentsRepository,
+  log
+} = {}) =>
+  Promise.resolve().then(() => {
```
[the remark this answers is not in the document]
Comment (alextremp) [reply src/cmp/infrastructure/cmpInitializer.js]: Yep :)

File: src/cmp/infrastructure/createEvent.js (hunk 1 above)
Comment (alextremp) [reply src/cmp/infrastructure/createEvent.js]: 👍 Let's see the polyfill stuff on next iterations when we'll need to package the dist version

**Comment by alextremp** [comment]
@victuxbb the requested changes are ready to be reviewed :)

**Pull request merged**